### PR TITLE
Readme

### PR DESCRIPTION
**Pull request by jlconlin**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+7. Generate a [CMake](https://cmake.org) file
+   ```bash
+   metaconfigure/generate.py cmake
+   ```
+8. Inspect resulting CMake file
+	 - Is the target of the expected type (executable, interface, or library)?
+	 - Are the tests specified?
+	 - Are the compiler flags specified?
+9. Build/test project locally to make sure everything works.
+10. Update `.travis.yml` and `.travis.sh`.
```
Comment (jlconlin) [inline README.md]: How do these need to be updated?

File: README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+   git subtree pull --squash --prefix metaconfigure https://github.com/njoy/metaconfigure.git master
+   ```
+4. Delete old `description.json` file.
+   ```bash
+   rm metaconfigure/description.json
+   ```
+5. Generate new `metaconfigure/description.json` file. This is done in Python.
+   ```python
+   import metaconfigure
+   args = {"name": "project-name", "language": "c++", "is external project": False, "include path": "src/include"}
```
Comment (jlconlin) [inline README.md:24]: We need to document all the arguments a user (namely me) cause use here.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+6. Inspect the new `metaconfigure/description.json`. In particular check:
+	- `implementation files`—ensure the correct relative path (to top-level of project) is correct.
+	- `header files`—ensure the correct relative path (to top-level of project) is correct.
+	- `include path`—ensure that this points to the correct path. This is used for projects external to NJOY such as [catch-adapter](https://github.com/njoy/catch-adapter).
+7. Generate a [CMake](https://cmake.org) file
+   ```bash
+   metaconfigure/generate.py cmake
+   ```
+8. Inspect resulting `CMakelists.txt` file
+	 - Is the target of the expected type (executable, interface, or library)?
```
Comment (jlconlin) [inline README.md]: Where should one look for these things? What should one look for?

Comment (apmccartney) [reply README.md]: Typically these appear near the end of the root source tree directory `CMakeLists.txt`. These types are described in the CMake documentation. An executable target will always be accompanied by a library or interface target in the CMake files generated by metaconfigure, so we'll start there. 

A library target is a collection of source files which are to be compiled into a library file (either static or dynamic based on configuration settings) during the build process. This implies that the project contains one or more implementation files (excluding a implementation file for any executable driver). Those declarations look something like

```
add_library( <name> ${<name>_library_linkage} <source files> )
```

(How's that for meta?)

For targets that lack implementation files (such as header only libraries), there's nothing to compile. As a historical note, prior to CMake 3.x, the only way to describe these libraries was using `add_custom_target` and specify several (undocumented) properties. In modern CMake, these targets are described using a "interface library" (which feels very kludgy, but oh well). These declarations look like

```
add_library( <name> INTERFACE <source files> )
```

Given an driver implementation file (defaulting to main.<extension> but can be specified in the description), you should also see an executable target which links to the library or interface target.
```
add_executable( <name>_executable <driver> )
set_target_properties( <name>_executable PROPERTIES OUTPUT_NAME <name> )
target_compile_options( <name>_executable PRIVATE <flags generator expression> )
target_link_libraries( <name>_executable <PUBLIC/INTERFACE> <name> )
```

File: README.md (hunk 2 above)
Comment (apmccartney) [reply README.md:24]: yeah, `metaconfigure`, in particular description generation, needs a manual. That's going to require a considerable investment of time. After this years deliverables are complete, I plan to return to documentation in the form of a LA-UR

File: README.md (hunk 1 above)
Comment (apmccartney) [reply README.md]: reference the corresponding files in Log.
https://github.com/njoy/Log/blob/master/.travis.yml
https://github.com/njoy/Log/blob/master/.travis/build.sh

**Pull request merged**